### ISSUE TITLE
Add optional subscription-manager fact generation

### ISSUE THREAD

**Issue by cgwalters** [containers/bootc#929]
We’re discussing subscription-manager integration with bootc. First step is change detection in bootc that can trigger a systemd unit.  
The main debate is where the writing of /etc/rhsm/bootc.facts lives. Proposal:

* Add a cargo feature `rhsm` to support enabling this code (want this to cover the Debian/Arch cases)  
* Add a systemd unit `bootc-rhsm-facts.service` triggered by our `bootc-status-updated.target. This unit should have ConditionDirectoryExists=/etc/rhsm (This dynamic detection will cover the fedora case)`  
* `The main thing to debate here is if we subpackage this as bootc-rhsm or just ship it in the main package. In a perfect world we’d subpackage it. But there are logistical overheads to doing this. Decision: Not a subpackage.`

**Comment by cgwalters** [comment]
It came up in chat if we could enable this feature in Fedora.

The background here is...Foreman/Satellite is a management tool that scrapes `/etc/rhsm` where some tools put data. My hope was that tooling learned to read bootc's API directly. This service acts effectively as another API.

That said, there's not really a substantial amount of code here, so I have no problems turning it on in Fedora.

The minor thing here is cosmetically everyone who doesn't have `subscription-manager` installed will see the service be skipped at startup. We could mitigate that with a generator instead...but eh.

**Comment by rriemann** [comment]
Indeed @cgwalters , I proposed Fedora bootc for a proof of concept deployment in the public sector. We elaborated the idea already early june at the [Hackdays 2025 in Paris](https://github.com/suitenumerique/hackdays2025/issues/16) We want to use Foreman fleetmanagement and I was surprised to see the fact files consumed by Foreman disabled for fedora.

The proof of concept will happen mostly in August and September. So if the bootc community wants to solve fact ingest to Foreman differently and enabling fact files takes anyway much longer to tickle down to Fedora Kinoite, then maybe it is not worth to rush it.
